pkg/types/aws/validation/machinepool.go

```diff
@@ -54,6 +58,40 @@ func ValidateMachinePool(platform *aws.Platform, p *aws.MachinePool, fldPath *fi
 
 	allErrs = append(allErrs, validateSecurityGroups(platform, p, fldPath)...)
 
+	if p.HostPlacement != nil {
+		allErrs = append(allErrs, validateHostPlacement(p, fldPath.Child("hostPlacement"))...)
+	}
+
+	return allErrs
+}
+
+func validateHostPlacement(p *aws.MachinePool, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if p.HostPlacement.Affinity != nil {
+		switch *p.HostPlacement.Affinity {
+		case aws.HostAffinityAnyAvailable:
+			if p.HostPlacement.DedicatedHost != nil {
+				allErrs = append(allErrs, field.Required(fldPath.Child("dedicatedHost"), "dedicatedHost is required when 'affinity' is set to DedicatedHost, and forbidden otherwise"))
+			}
+		case aws.HostAffinityDedicatedHost:
+			if p.HostPlacement.DedicatedHost == nil {
```

Comment:
```suggestion
			if len(p.HostPlacement.DedicatedHost) == 0 {
```

Both `nil` and empty slice are invalid, right?